jax/_src/lax/lax.py

```diff
@@ -966,12 +966,45 @@ class GatherDimensionNumbers(NamedTuple):
   start_index_map: Sequence[int]
 
 
+class GatherScatterMode(IntEnum):
+  # Indices will be clamped to the nearest in-range value, i.e., such that the
+  # entire window to be gathered is in-range.
+  CLIP = 0
+
+  # If any part of a gathered window is out of bounds, the entire window
+  # that is returned, even those elements that were otherwise in-bounds, will be
+  # filled with a constant.
+  # If any part of a scattered window is out of bounds, the entire window
+  # will be discarded.
+  FILL_OR_DROP = 1
+
+  # The user promises that indices are in bounds. No additional checking will be
+  # performed. In practice, with the current XLA  implementation this means
+  # that, out-of-bounds gathers will be clamped but out-of-bounds scatters will
+  # be discarded. Gradients will not be correct if indices are out-of-bounds.
+  PROMISE_IN_BOUNDS = 2
+
+  @staticmethod
+  def from_any(s: Optional[Union[str, 'GatherScatterMode']]):
+    if s == "clip" or s == GatherScatterMode.CLIP:
+      return GatherScatterMode.CLIP
+    if s == "fill" or s == "drop" or s == GatherScatterMode.FILL_OR_DROP:
+      return GatherScatterMode.FILL_OR_DROP
+    if (s is None or s == "promise_in_bounds" or
+        s == GatherScatterMode.PROMISE_IN_BOUNDS):
+      return GatherScatterMode.PROMISE_IN_BOUNDS
```

Comment:
`isintance` on the enum might help simplify this:
```suggestion
    if isinstance(s, GatherScatterMode):
      return s
    if s == "clip":
      return GatherScatterMode.CLIP
    if s == "fill" or s == "drop":
      return GatherScatterMode.FILL_OR_DROP
    if s is None or s == "promise_in_bounds":
      return GatherScatterMode.PROMISE_IN_BOUNDS
```

Reply:
Done.